src/sentry/integrations/slack/message_builder/footer.py

```diff
@@ -0,0 +1,80 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Sequence
+
+from sentry import roles
+from sentry.models import Environment, Group, OrganizationMember, Team, User
+from sentry.notifications.notifications.base import BaseNotification
+from sentry.notifications.utils.urls import get_settings_url
+from sentry.types.integrations import ExternalProviders
+
+# Defined as a variable so that other providers can more easily copy this file.
+PROVIDER = ExternalProviders.SLACK
+
+
+@dataclass
+class FooterLink:
+    label: str
+    url: str
+
+    SETTINGS_URL = object()
+
+
+def build_link(link: FooterLink, provider: ExternalProviders) -> str:
+    if provider == ExternalProviders.SLACK:
+        return f"<{link.url}|{link.label}>"
+    raise RuntimeError("Invalid Provider")
+
+
+def process_footer_part(
+    notification: BaseNotification,
+    recipient: Team | User,
+    provider: ExternalProviders,
+    part: FooterLink | str,
```

Comment:
If `part` is a link can you call this variable `link`?

Reply:
`part` can be a string or a link.